### PR TITLE
neural network from scratch with gradient descent

### PR DESCRIPTION
**Pull request by Sairamasish**
The python file attached serves as template. Current template is mainly designed for computer vision tasks.The dataset (.h5 files) is not added and it is purely dependent on user(like can be used for cats and dogs, pneumonia vs non pneumonia,etc.).

### CODE REVIEW

**Review by harshareddy794 — APPROVED** [review]
That's really awesome @Sairamasish 
Thanks for your contribution 
btw you can make another PR for datasets also
